narwhal/crypto/src/hash.rs

```diff
@@ -1,17 +1,16 @@
+use crate::CurrentNetwork;
+
 use std::fmt;
 
 use base64ct::{Base64, Encoding};
 use bitvec::prelude::{BitVec, Lsb0};
-
 use proptest_derive::Arbitrary;
-
 use serde::{Deserialize, Serialize};
-use snarkvm_console::{
-    algorithms::BHP256, network::Testnet3, prelude::Hash as _, prelude::ToBytes,
-};
+use snarkvm_console::{algorithms::BHP256, prelude::Hash as _, prelude::ToBytes};
 
-type CurrentNetwork = Testnet3;
+pub const DIGEST_LENGTH: usize = snarkvm_console::types::Field::<CurrentNetwork>::SIZE_IN_BYTES;
```

Comment:
is this still necessary? I saw you delete this generic everywhere?

Reply:
It's used in tests but I _think_ I might be able to get rid of it (and then it'll just be private)

Reply:
ah, actually I can't, it's used to seed proptest vectors